### PR TITLE
Update Freddy to use latest Hare

### PR DESCRIPTION
**Pull request by take-five**
Changes in Hare include:
* Support for unroutable messages in RPC.Client
* Improved reconnection mechanism

Also `Freddy.Consumer.cancel/1` is removed. Automatic graceful shutdown is to be implemented later.

### CODE REVIEW

**Comment by deiwin** [comment]
Hare diff, please?

**Comment by take-five** [comment]
@deiwin https://github.com/jcabotc/hare/pull/13

File: test/freddy/consumer_test.exs [hunk 1]
```diff
@@ -136,20 +136,6 @@ defmodule Freddy.ConsumerTest do
       tear_down(consumer)
     end
 
-    test "consumer stops after receives cancel message", %{conn: conn, history: history} do
```
Comment (deiwin) [inline test/freddy/consumer_test.exs]: Why was this removed?

File: mix.exs [hunk 2]
```diff
@@ -36,8 +36,9 @@ defmodule Freddy.Mixfile do
   # Type "mix help deps" for more examples and options
   defp deps do
     [
-      {:hare, git: "https://github.com/take-five/hare"},
-      {:amqp, "~> 0.2.0"},
+      # TODO: change when https://github.com/jcabotc/hare/pull/13 is merged
+      {:hare, git: "https://github.com/salemove/hare", branch: "improve-reconnection"},
```
Comment (deiwin) [inline mix.exs]: This should probably be updated, right?

Comment (take-five) [inline mix.exs]: Yeah, I'll update as soon as `hare` is published to hex.pm

File: test/freddy/consumer_test.exs (hunk 1 above)
Comment (take-five) [inline test/freddy/consumer_test.exs]: 1. Manual shutdown doesn't add much value except removing some noise from the logs
2. Manual shutdown is very hard to setup
3. To perform cleanup we can use "cleanup crew" pattern as described here: http://erlang.org/pipermail/erlang-questions/2013-March/072693.html (TL;DR - spawn a side process that automatically closes AMQP channel when consumer dies). Not now though.
4. **OR** we can trap EXIT signals and let GenServer execute `terminate/2` callback which will close the channel

File: mix.exs (hunk 2 above)
Comment (take-five) [inline mix.exs]: @deiwin updated

File: mix.exs [hunk 3]
```diff
@@ -36,8 +36,8 @@ defmodule Freddy.Mixfile do
   # Type "mix help deps" for more examples and options
   defp deps do
     [
-      {:hare, git: "https://github.com/take-five/hare"},
-      {:amqp, "~> 0.2.0"},
+      {:hare, "~> 0.2.0", hex: :salemove_hare},
+      {:amqp, "~> 0.2.2"},
```
Comment (deiwin) [inline mix.exs]: Why are we specifying `amqp` directly? Do we also use it directly?

Comment (take-five) [inline mix.exs]: In Hare `amqp` dependency is optional. But in fact we always gonna use it, so here we just declare it as required. No, we don't use it directly.
